src/libraries/System.DirectoryServices.Protocols/tests/LdapConnectionTests.cs

```diff
@@ -291,6 +293,18 @@ public void Dispose_MultipleTimes_Nop()
             connection.Dispose();
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void CanCreateConcurrentConnections()
+        {
+            RemoteExecutor.Invoke(() =>
```

Comment:
Just so I understand you only use RemoteExecutor since you don’t want other tests to run first and instantiate it for you right?

Also even if we do this I wonder if we can get a repro fairly easily si ce it is obviously still possible for some thread to execute a bit sooner than others causing the issue not to repro

Reply:
> Just so I understand you only use RemoteExecutor since you don’t want other tests to run first and instantiate it for you right?

yep.

> Also even if we do this I wonder if we can get a repro fairly easily si ce it is obviously still possible for some thread to execute a bit sooner than others causing the issue not to repro

True, I'm doing this first to make sure it's good enough to repro occasionally in CI.  It's not full proof, race conditions never are.  If I can't get a repro I'll make it more agressive (running RE multiple times, for example, or spinning up multiple threads having them wait on event, then signalling).  My thinking is that if we regress here this will start crashing flakily and we'll notice it.  It should be a pretty good indicator, and better than our current crash which has taken us quite a while to track down (though we still did despite its flakiness).

Reply:
Seems like it didn’t repro, so we probably do need a more aggressive concurrency approach

Reply:
Potentially.  I've been trying locally and haven't had any success, as least in WSL.  It's possible that even the crash is more nuanced than we can reliably reproduce.  I'll try tossing up my test case into Helix and see if its any luckier than I have been in WSL.

Reply:
Frustrating, I still can't repro the crash.  The dumps we have are clear enough that show two threads in ldap_init at the same time, but I can't seem to force it.  😠

Reply:
You may need to make sure that:
- The other tests finished running so that all processors are ready to go. `Thread.Sleep(60000)` as the first thing in the test may do it.
- Use spinloop on `static volatile bool` variable to kick off all threads. It is the most precise way to get all threads going at the exact same time.

Reply:
I'll let this run finish then toss up another attempt to see if spin wait is any better.

Reply:
Hmm, I do not think that the spin wait is going to help. The threads will get blocked on JIT lock multiple times before they reach the pinvoke. Waiting a bit for the machine to be idle and/or retrying for fixed amount of time instead of fixed number of iterations may work better.

Reply:
Ok. I’ll give that a try